scripts/py_matter_idl/matter_idl/generators/java/__init__.py

```diff
@@ -70,7 +71,54 @@ def FieldToGlobalName(field: Field, context: TypeLookupContext) -> Union[str, No
     return None
 
 
-def CallbackName(attr: Attribute, cluster: Cluster, context: TypeLookupContext) -> str:
+def GlobalNameToJavaName(name: str) -> str:
+    if name in {'Int8u', 'Int8s', 'Int16u', 'Int16s'}:
+        return 'Integer'
+
+    if name.startswith('Int'):
+        return 'Long'
+
+    # Double/Float/Booleans/CharString/OctetString
+    return name
+
+
+def DelegatedCallbackName(attr: Attribute, context: TypeLookupContext) -> str:
+    """
+    Figure out what callback name to use when a variable requires a read callback.
+
+    These are split into native types, like Boolean/Float/Double/CharString, where
+    one callback type can support anything.
+
+    For specific types (e.g. A struct) codegen will generate its own callback name
+    specific to that type.
+    """
+    global_name = FieldToGlobalName(attr.definition, context)
+
+    if global_name:
+        return 'Delegated{}AttributeCallback'.format(GlobalNameToJavaName(global_name))
+
+    return 'Delegated{}Cluster{}AttributeCallback'.format(context.cluster.name, capitalcase(attr.definition.name))
+
+
+def ChipClustersCallbackName(attr: Attribute, context: TypeLookupContext) -> str:
+    """
```

Comment:
Doc string seems like a copy and paste and doesn't seem to be correct

Reply:
Updated. Unfortunately we have 3 very similar methods, all computing "callback names" but under different usages (C++ for one, java for 2 of them). I am not actually familiar enough with the intent of the codegen to more intelligently explain the difference, so left it quite vague.